synapse/event_auth.py

```diff
@@ -471,14 +471,14 @@ def _check_power_levels(event, auth_events):
     ]
 
     old_list = current_state.content.get("users", {})
-    for user in set(old_list.keys() + user_list.keys()):
+    for user in set(list(old_list.keys()) + list(user_list.keys())):
         levels_to_check.append(
             (user, "users")
         )
 
     old_list = current_state.content.get("events", {})
     new_list = event.content.get("events", {})
-    for ev_id in set(old_list.keys() + new_list.keys()):
+    for ev_id in set(list(old_list.keys()) + list(new_list.keys())):
```

Comment:
If we're doing `list(foo.keys())` can't we just do `list(foo)`? Rather than in py2 have the lists be copied twice

Reply:
Good idea, done.